### PR TITLE
CIF-2120: Add StorefrontEvents to PDP

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

The product component generates the context required by Magento Storefront Events SDK

## Related Issue

CIF-2120

## Motivation and Context

Required by the product recommendations engine

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/cif/clientlibs/storefront-events/js/collector.js [hunk 1]
```diff
@@ -17,3 +17,18 @@ import mse from '@adobe/magento-storefront-events-sdk';
 
 // Expose Magento Storefront Events SDK on the global window object
 window.magentoStorefrontEvents = mse;
+
+const processProductStorefrontData = () => {
```
Comment (herzog31) [inline ui.apps/src/main/content/jcr_root/apps/core/cif/clientlibs/storefront-events/js/collector.js:21]: We should wrap this part into an IIFE and also check for the DOM to be ready. Similar to https://github.com/adobe/aem-core-wcm-components/blob/development/content/src/scripts/datalayer/v1/datalayer.js#L100-L104.

**Review by herzog31 — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/storefrontcontext/ProductStorefrontContext.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.storefrontcontext;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public interface ProductStorefrontContext extends CommerceStorefrontContext {
+
+    @JsonProperty("productId")
+    Integer getId();
```
Comment (mhaack) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/storefrontcontext/ProductStorefrontContext.java:21]: I assume id being an Integer is given by the events format? Can we make it String also to be future ready with UID and other ID formats?

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    private final ProductInterface product;
+
+    public ProductStorefrontContextImpl(ProductInterface product) {
+        this.product = product;
+    }
+
+    @Override
+    public Integer getId() {
+        return 0;
```
Comment (mhaack) [inline bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java:29]: ? We ignore the actual product id?

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/storefrontcontext/ProductStorefrontContext.java (hunk 2 above)
Comment (herzog31) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/storefrontcontext/ProductStorefrontContext.java:21]: It has to be an integer according to the Event SDK schema: https://github.com/adobe/magento-storefront-events-sdk/blob/main/src/types/schemas/product.ts#L7.

File: bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java (hunk 3 above)
Comment (herzog31) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java:29]: We have confirmation that internally SKU is used to match products. The id however is still a mandatory field in the schema. So as long as UID support is missing, we decided to just go with a dummy value that fulfils the schema.

Comment (mhaack) [reply bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/storefrontcontext/ProductStorefrontContextImpl.java:29]: Ok maybe add a //TODO comment about this behavior and that we work around bad Events SDK design.

**Review by mhaack — APPROVED** [review]

**Pull request merged**